Pod/Classes/SEGMoEngageIntegration.m

```diff
@@ -25,47 +25,79 @@ - (void)track:(SEGTrackPayload *)payload
 
 - (void)identify:(SEGIdentifyPayload *)payload
 {
-    NSDictionary *traits = payload.traits;
-
+    NSDictionary *moengagePayloadDict = [payload.traits copy];
+    
+    NSMutableDictionary *traits = [NSMutableDictionary dictionaryWithDictionary:moengagePayloadDict];
+    if(![traits.allKeys count]){
```

Comment:
> An empty dict will have no keys.

Hence it's count will be zero — and `traits.count` would be sufficient. Here's a sample to show the behaviour.

```
        if (@{}.count) {
            NSLog(@"@{}.count is truthy");
        } else {
            NSLog(@"@{}.count is falsy");
        }
        NSDictionary *dict;
        if (dict.count) {
            NSLog(@"nil.count is truthy");
        } else {
            NSLog(@"nil.count is falsy");
        }
```

In both cases, the `else` block is executed.

I'm not saying what you have doesn't work — it's just unnecessary to read they keys and then check the count since `traits.count` should always be the same as `traits.allKeys.count`.

I might be missing something, can you share an example case where `traits.count != traits.allKeys.count`?